### PR TITLE
docs: add information on using a custom regexp in `definePageMeta`

### PR DESCRIPTION
**Pull request by L422Y**
Adds an example of using a path matcher and links to more information on the matching syntax

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Adds extended information for using a path matcher and links to more information on the matching syntax, links to the example from the `path` configuration option definition. Allows devs without previous knowledge of advanced path matching syntax to preview what is possible with the `path` option in `definePageMeta`

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/3.api/3.utils/define-page-meta.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ interface PageMeta {
 
   - **Type**: `string`
 
-    You may define a path matcher, if you have a more complex pattern than can be expressed with the file name.
+    You may define a [path matcher](#using-an-advanced-path-matcher), if you have a more complex pattern than can be expressed with the file name.
```
Comment (danielroe) [inline docs/3.api/3.utils/define-page-meta.md:63]:
```suggestion
    You may define a [custom regular expression](#using-a-custom-regular-expression), if you have a more complex pattern than can be expressed with the file name.
```

File: docs/3.api/3.utils/define-page-meta.md [hunk 2]
```diff
@@ -183,6 +183,24 @@ definePageMeta({
 </script>
 ```
 
+### Using an advanced path matcher
```
Comment (danielroe) [inline docs/3.api/3.utils/define-page-meta.md:186]:
```suggestion
### Using a Custom Regular Expression
```

File: docs/3.api/3.utils/define-page-meta.md [hunk 3]
```diff
@@ -183,6 +183,24 @@ definePageMeta({
...
+
+A path matcher is a good way to resolve conflicts between overlapping routes, for instance:
```
Comment (danielroe) [inline docs/3.api/3.utils/define-page-meta.md:188]:
```suggestion
A custom regular expression is a good way to resolve conflicts between overlapping routes, for instance:
```

**Review by danielroe — APPROVED** [review]

File: docs/3.api/3.utils/define-page-meta.md [hunk 4]
```diff
@@ -183,6 +183,24 @@ definePageMeta({
 </script>
 ```
 
+### Using a Custom Regular Expression
+
+A custom regular expression is a good way to resolve conflicts between overlapping routes, for instance:
+
+The two routes "/test-category" and "/1234-post" match both `[postId]-[postSlug].vue` and `[categorySlug].vue` page routes.
+
+In order to make sure that we are only matching digits (`\d+`) for `postId` in the `[postId]-[postSlug]` route, we can add the following to the `[postId]-[postSlug].vue` page template:
```
Comment (DamianGlowala) [inline docs/3.api/3.utils/define-page-meta.md:192]: more concise/less wordy
```suggestion
To make sure that we are only matching digits (`\d+`) for `postId` in the `[postId]-[postSlug]` route, we can add the following to the `[postId]-[postSlug].vue` page template:
```

**Review by DamianGlowala — CHANGES_REQUESTED** [review]

**Review by DamianGlowala — APPROVED** [review]

**Pull request merged**